review_routes/v3/results_routes.py

```diff
@@ -607,6 +603,138 @@ async def build_compare_results_main_query(
     return main_assessment_query, total_rows
 
 
+async def build_missing_words_main_query(
+    revision_id: Optional[int],
+    reference_id: Optional[int],
+    book: Optional[str],
+    chapter: Optional[int],
+    verse: Optional[int],
+    page: Optional[int],
+    page_size: Optional[int],
+    db: Session,
+) -> Tuple:
+    if page is not None and page_size is not None:
+        offset = (page - 1) * page_size
+        limit = page_size
+
+    else:
+        offset = 0
+        limit = None
+
+    main_assessment = (
+        db.query(Assessment)
+        .filter(
+            Assessment.revision_id == revision_id,
+            Assessment.reference_id == reference_id,
+            Assessment.type == "word-alignment",
+            Assessment.status == "finished",
+        )
+        .order_by(Assessment.end_time.desc())
+        .first()
+    )
+    if not main_assessment:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="No completed assessment found for the given revision_id and reference_id",
+        )
+    main_assessment_id = main_assessment.id
+    main_assessment_query = (
+        db.query(
+            AlignmentTopSourceScores.id.label("id"),
+            AlignmentTopSourceScores.book.label("book"),
+            AlignmentTopSourceScores.chapter.label("chapter"),
+            AlignmentTopSourceScores.verse.label("verse"),
+            AlignmentTopSourceScores.source.label("source"),
+            AlignmentTopSourceScores.score.label("score"),
+        )
+        .filter(
+            AlignmentTopSourceScores.assessment_id == main_assessment_id,
+            AlignmentTopSourceScores.score < 0.15,
+        )
+        .order_by("id")
+    )
+    if book:
+        main_assessment_query = main_assessment_query.filter(
+            AlignmentTopSourceScores.book == book
+        )
+    if chapter:
+        main_assessment_query = main_assessment_query.filter(
+            AlignmentTopSourceScores.chapter == chapter
+        )
+    if verse:
+        main_assessment_query = main_assessment_query.filter(
+            AlignmentTopSourceScores.verse == verse
+        )
+
+    total_rows = main_assessment_query.count()
+    main_assessment_query = main_assessment_query.limit(limit).offset(offset)
+
+    return main_assessment_query, total_rows, main_assessment_id
+
+
+async def build_missing_words_baseline_query(
```

Comment:
can we add a docstring to both fuctions, i think i understand the purpose but for future reference